### ISSUE TITLE
Add Extension Method for adding switches to ProcessArgumentBuilder

### ISSUE THREAD

**Issue by gep13** [cake-build/cake#453]
In another [PR](https://github.com/cake-build/cake/pull/346), @RichiCoder1 did some work to add some additional extension methods to make the creation of switches and argument easier to add to the ProcessArgumentBuilder.

This issue is to track the inclusion of that work, as it was removed from the GitVersion PR as that was a different scope of work.

So, instead of doing this:

```
builder.Append("-c");
builder.AppendQuoted(settings.CacheLocation);
```

You would be able to use:

```
builder.AppendSwitchQuoted("--version", settings.Version);
```

**Comment by RichiCoder1** [comment]
I forgot about that haha

**Comment by gep13** [comment]
I removed the work from your other PR, but I have saved it, and didn't want it to get lost.  Creating a PR to include it just now, if you have no objections?

**Comment by RichiCoder1** [comment]
None at all.

**Comment by gep13** [comment]
This was released in Cake 0.6.0: http://cakebuild.net/blog/2015/11/cake-v0-6-0-released
